SU2_CFD/src/solvers/CDiscAdjSolver.cpp

```diff
@@ -261,6 +259,8 @@ void CDiscAdjSolver::RegisterVariables(CGeometry *geometry, CConfig *config, boo
     config->SetIncPressureOut_BC(BPressure);
     config->SetIncTemperature_BC(Temperature);
 
+    direct_solver->RegisterSolutionExtra(reset, config);
```

Comment:
I don't think the boolean passed to this function means "reset". But I'll assign @TobiKattmann to this one.

Reply:
`unsigned long CIncEulerSolver::RegisterSolutionExtra(bool input, const CConfig* config) {
  if (config->GetKind_Streamwise_Periodic() == ENUM_STREAMWISE_PERIODIC::MASSFLOW) {
    if (input) AD::RegisterInput(SPvals.Streamwise_Periodic_PressureDrop);
    else AD::RegisterOutput(SPvalsUpdated.Streamwise_Periodic_PressureDrop);
    return 1;
  }
  return 0;
}`

Yes, you are right, so we are doing the opposite then. I will investigate a bit more.